openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1507,7 +1507,96 @@ It is up to the Credential Issuer whether to update both the signature and the c
+containing privacy-sensitive data only for as long as needed, including in log files. Any logging of
+End-User data should be carefully considered as to whether it is necessary at all. The time logs are retained
+for should be minimized.
+
+After Issuance, Credential Issuers SHOULD NOT store the Issuer-signed Credentials if they
+contain privacy-sensitive data. Wallets SHOULD store Credentials only in encrypted form, and,
+wherever possible, use hardware-backed encryption. Credentials that are no longer valid to present should be deleted as soon as
+possible.
+
+## Correlation 
+
+### Unique Values Encoded in the Credential
+
+Issuance/presentation or two presentation sessions by the same End-User can be linked on the basis of
+unique values encoded in the Credential (End-User claims, identifiers, Issuer signature, etc.) either by colluding Issuer/Verifier or Verifier/Verifier pairs, or by the same Verifier.
+
+To prevent these types of correlation, Credential Issuers and Wallets SHOULD use
+methods, including but not limited to the following ones:
+
+* Issue a batch of Credentials to enable the usage a unique Credential per presentation or per Verifier using Batch Credential Endpoint defined in (#batch-credential-endpoint). This only helps with Verifier/Verifier unlinkability.
+* Use cryptographic schemes that can provide non-correlation.
+
+### Credential Offer
+
+The Privacy Considerations in Section 11.2 of [@!RFC9101] apply to the `credential_offer` and
+`credential_offer_uri` parameters defined in (#credential-offer).
+
+### Authorization Request
+
+The Wallet SHOULD NOT include potentially sensitive information in the Authorization Request,
+for example, that including clear-text session information as a `state` parameter value, or encoding
+it in a `redirect_uri` parameter. A third party may observe such information through browser
+history, etc. and correlate the user's activity using it.
+
+## Credential Issuer Identifier
+
+The Credential Issuer Identifier alone may reveal information about the End-User.
+
+For example, when a military organization or a drug rehabilitation center issues a vaccine
+credential, verifiers can deduce that the owner of the Wallet storing such Credential is a
+military member or may have a substance use disorder.
+
+In addition, when a Credential Issuer issues only one type of Credential, it might have privacy implications,
+because if the Wallet has a Credential issued by that Issuer, its type and claim names can be
+determined.
+
+For example, if the National Cancer Institute only issued Credentials with cancer registry
+information, it is possible to deduce that the owner of the Wallet storing such Credential is a
+cancer patient.
+
+To mitigate this issue, a group of Credential Issuers may elect to use a common Credential Issuer
+Identifier. A group signature scheme may also be used, instead of an individual signature.
```

Comment:
Is this about the `Credential Issuer Identifier`, a metadata field in OpenID4VCI, or about generally linking credentials to issuers using identifiers in the credential?

If it is the former, how does the `Credential Issuer Identifier` become shared with a verifier?

Reply:
> If it is the former, how does the Credential Issuer Identifier become shared with a verifier?

It is the former. In an SD-JWT VC for example, the credential issuer identifier is stored in the `iss` claim in the credential send to the verifier.

Reply:
`mdoc` does not use `Credential Issuer Identifier`, but it does have a certificate that identifies the issuer. So I think this section needs to be more general, and talk about any field in the credential that could identify a specific issuer, and therefore disclose more information than intended.

Reply:
Suggested replacement text for this section:

> ## Credential Issuer identifiers
>
> Information in the credential identifying a particular Credential Issuer, such as an issuer identifier or certificate, may reveal information about the End-User.
> 
> For example, when a military organization or a drug rehabilitation center issues a vaccine credential, verifiers can deduce that the owner of the Wallet storing such a credential is a military member or may have a substance use disorder.
>   
> In addition, when a Credential Issuer issues only one type of credential, it might have privacy implications because if the Wallet has a credential issued by that Issuer, its type and claim names can be determined.
> 
> For example, if the National Cancer Institute only issued credentials with cancer registry information, it is possible to deduce that the owner of the Wallet storing such a credential is a cancer patient.
> 
> To mitigate these issues, a group of organizations may elect to share a single Credential Issuer, such that any credentials issued by this Issuer cannot be attributed to a particular organization through issuer identifiers alone. A group signature scheme may also be used instead of an individual signature.

Reply:
> Suggested replacement text for this section:
> 
> > ## Credential Issuer identifiers
> > Information in the credential identifying a particular Credential Issuer, such as an issuer identifier or certificate, may reveal information about the End-User.
+1

> > To mitigate these issues, a group of organizations may elect to share a single Credential Issuer, such that any credentials issued by this Issuer cannot be attributed to a particular organization through issuer identifiers alone. A group signature scheme may also be used instead of an individual signature.

This is a more elegant approach but I would not rule out use of a technique described in the original sentence i.e. single issuer identifier shared across multiple Issuers.

Reply:
@cobward 's suggestion seems sensible to me, though I think I would avoid using the term 'Credential Issuer identifiers' or 'issuer identifiers' as this seems highly likely to be confused with the defined term 'Credential Issuer Identifier'.

Perhaps something like "User-group specific Credential Issuers" for the title?

I think we should mention 'public keys' in the "such as" in the first paragraph too.

Reply:
@cobward thank you for the suggestion - I made the suggestion here: https://github.com/openid/OpenID4VCI/pull/244/files#r1480497886

Reply:
> I think we should mention 'public keys' in the "such as" in the first paragraph too.

+1

> This is a more elegant approach but I would not rule out use of a technique described in the original sentence i.e. single issuer identifier shared across multiple Issuers.

@yaromin I believe a shared issuer identifier alone does not solve the problem. As @jogu highlighted, a public key alone is enough to identify an issuer.